src/MultiAgentReinforcementLearning/src/MultiAgentReinforcementLearning.jl

```diff
@@ -0,0 +1,224 @@
+module MultiAgentReinforcementLearning
+
+export MultiAgentPolicy
+export MultiAgentHook
+
+using ReinforcementLearningBase
+using ReinforcementLearningCore
+
+import ReinforcementLearningCore: RLCore
+import Base.getindex
+import Base.iterate
+
+struct MultiAgentPolicy <: AbstractPolicy
+    agents::NamedTuple
+
+    function MultiAgentPolicy(agents::NamedTuple)
+        new(agents)
+    end
+end
+
+struct MultiAgentHook <: AbstractHook
+    hooks::NamedTuple
+
+    function MultiAgentHook(hooks::NamedTuple)
+        new(hooks)
+    end
+end
+
+
+(p::MultiAgentPolicy)(env::AbstractEnv) = nothing # Default does nothing, but might be useful for some environments to clean up / pass final state to agents
+
+struct CurrentPlayerIterator
+    env::AbstractEnv
+end
+
+function current_player_iterator(env)
```

Comment:
I would avoid creating a second constructor unless there's a specific reason to do it.

Reply:
Great point!